internal/api/middleware.go

```diff
@@ -152,6 +148,45 @@ func authMiddleware(ctx *middlewareContext) middleware {
 	}
 }
 
+func getClaimsFromAuthorizationHeader(header string, jwtSecret []byte) (*jwtGocertClaims, error) {
+	if header == "" {
+		return nil, fmt.Errorf("authorization header not found")
+	}
+	bearerToken := strings.Split(header, " ")
+	if len(bearerToken) != 2 || bearerToken[0] != "Bearer" {
+		return nil, fmt.Errorf("authorization header couldn't be processed. The expected format is 'Bearer <token>'")
+	}
+	claims, err := getClaimsFromJWT(bearerToken[1], jwtSecret)
+	if err != nil {
+		return nil, fmt.Errorf("token is not valid: %s", err)
+	}
+	return claims, nil
+}
+
+func permitter(claims *jwtGocertClaims, regex, path string, SelfAuthorizedAllowed bool) (pathMatched, idMatchedIfExistsInPath bool, err error) {
```

Comment:
This whole function is a bit unclear in what it is supposed to do at a glance. The name is quite generic, and it seems to return booleans that things matched in the accessed path, and not permit or deny anything.

I think the name could be improved and a docstring would be helpful here.

Reply:
reworked it a bit more. To be honest this entire thing could be improved a lot but it's difficult to avoid writing a new go web framework without having the code be ugly and difficult to read.